### ISSUE TITLE
Create simple database structure

### ISSUE THREAD

**Issue by manwar** [manwar/theweeklychallenge#123]
I am looking for simple database to hold user data and contribution stats. 

Just simple SQL file to create database, table and constraints where needed.

Please share your ideas before you start working on it,

**Comment by NoubissiViany** [comment]
@manwar  Certainly! I can help you design a simple database to hold user data and contribution stats. Here's a basic outline of the structure:

Database: UserContributions

Table: Users

Columns:
user_id (integer, primary key)
username (varchar, unique)
email (varchar)
password (varchar)
created_at (datetime)
Table: Contributions

Columns:
contribution_id (integer, primary key)
user_id (integer, foreign key referencing Users.user_id)
contribution_date (date)
contribution_amount (decimal)

Please are we going to create this database structure using Perl or Raku??? Also, can you please assign this task to me?

**Comment by manwar** [comment]
@NoubissiViany Everything we do here would be Perl only. 
In my opinion we would need the following tables at the most:

a) users: user_id, user_type, user_password
b) user_types: user_type_id, user_type_name
c) languages: language_id, language_name
d) user_contributions: user_id, language_id, week_id, count
e) weeks: week_id, week_name

**Comment by NoubissiViany** [comment]
@manwar Alright!!! can you please assign this task to me???

**Comment by manwar** [comment]
@NoubissiViany few more to add

permissions: permission_id, permission_name
roles: role_id, permission_id
groups: group_id, role_id

**Comment by NoubissiViany** [comment]
@manwar alright sir well received

**Comment by manwar** [comment]
@NoubissiViany another addition as below

task_tags: task_tag_id. task_tag_name
tasks: week_id, task_tag, task_description, is_active

**Comment by NoubissiViany** [comment]
@manwar okay well received

**Comment by manwar** [comment]
@NoubissiViany another one for you:

champions: champion_id, user_id, champion_title
interview_questions: question_id, question_text, is_active
interview_responses: champion_id, question_id, question_response

**Comment by manwar** [comment]
@NoubissiViany I was wondering if you could create some kind of ERD so that it is easier to visualise the relationships?

I found this which offers FREE plan. Please take a look if it is of any use.
https://creately.com/lp/er-diagram-tool-online/

**Comment by NoubissiViany** [comment]
@manwar Okay I'm going to take a look at it when I am done with the updates of the data structure, I need to be done with it so that the person working on issue #124 can start working
